client/src/pages/TaskHistoryPage/TaskHistoryPage.js

```diff
@@ -1,23 +1,71 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import styles from "./TaskHistoryPage.module.css";
 import TaskListView from "../../components/TaskListView/TaskListView.js";
+import {
+  buildDailyTaskRoute,
+  buildDateRangeRoute,
+  convertTaskData,
+  getWeekRange,
+  getMonthRange,
+} from "../../utils.js";
+import { ErrorMessage } from "../../components/ErrorMessage/ErrorMessage.js";
+import { instance } from "../../axios.js";
 
 const TaskHistoryPage = () => {
-  const sampleTask = {
-    title: "gdsc meeting",
-    location: "dh2020",
-    startTime: "05:00pm",
-    endTime: "07:00pm",
-    isOngoing: false,
+  const dayFilter = "day";
+  const weekFilter = "week";
```

Comment:
At least for the current intended behaviour, this wouldn't work because some weeks span two different months. For example, the week of April 1 (this week) starts at Monday March 28 and ends at Sunday April 3. The month call would only return tasks from April 1 - April 30.

Reply:
Ah I see. For an edge case such as this, we'll need to end up making 2 API calls. (Lets leave the week filter for now and we'll mainly focus on the day and month filtration first)